pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ConsumedLedgersTrimTest.java

```diff
@@ -150,8 +150,8 @@ public void testConsumedLedgersTrimNoSubscriptions() throws Exception {
         pulsar.getAdminClient().topics().getStats(topicName);
         MessageId messageIdAfterRestart = pulsar.getAdminClient().topics().getLastMessageId(topicName);
         LOG.info("lastmessageid " + messageIdAfterRestart);
-        assertEquals(messageIdAfterRestart, messageIdBeforeRestart);
-
+        Awaitility.await().untilAsserted(()->
+                assertEquals(messageIdAfterRestart, messageIdBeforeRestart));
```

Comment:
The `messageIdAfterRestart` and `messageIdBeforeRestart` will never be changed when you retry the condition by using Awaitility, so it will not fix the flaky test but increase the test time while `messageIdAfterRestart` not equals `messageIdBeforeRestart`

Reply:
Thanks to your review, I will try to fix it again.